cpp/src/bitmask/null_mask.cu

```diff
@@ -91,30 +91,26 @@ namespace {
  * @param[out] global_count The number of non-zero bits in the specified range
  *---------------------------------------------------------------------------**/
 template <size_type block_size>
-__global__ void count_set_bits_kernel(bitmask_type const*bitmask,
+__global__ void count_set_bits_kernel(bitmask_type const *bitmask,
                                       size_type start_bit_index,
                                       size_type stop_bit_index,
                                       size_type *global_count) {
   constexpr auto const word_size{detail::size_in_bits<bitmask_type>()};
 
   auto const start_word_index{word_index(start_bit_index)};
   auto const stop_word_index{word_index(stop_bit_index)};
-
-  // There's always at least one word to count
-  auto const num_words{thrust::max((stop_word_index - start_word_index), 1)};
-
-  size_type tid = threadIdx.x + blockIdx.x * blockDim.x;
+  auto const tid = threadIdx.x + blockIdx.x * blockDim.x;
+  auto thread_word_index = tid + start_word_index;
   size_type thread_count{0};
 
   // First, just count the bits in all words
-  while (tid < num_words) {
+  while (thread_word_index <= stop_word_index) {
     thread_count += __popc(bitmask[tid + start_word_index]);
```

Comment:
You didn't update this line to use `thread_word_index` instead of `tid`. Shouldn't it be:

```suggestion
     thread_count += __popc(bitmask[thread_word_index]);
```

Reply:
Good catch. It didn't show up in any tests because the grid size I'm launching doesn't actually use the grid stride loop.